CharlieBackend.Data/Repositories/Impl/LessonRepository.cs

```diff
@@ -33,6 +33,12 @@ public async Task<List<Lesson>> GetAllLessonsForMentor(long mentorId)
                 .ToListAsync();
         }
 
+        public async Task<Lesson> GetLessonByHomeworkId(long homeworkId)
+        {
+            return await _applicationContext.Lessons
+                .FirstOrDefaultAsync(l => l.Id == _applicationContext.Homeworks.FirstOrDefault(h => h.Id == homeworkId).LessonId);
+        }
+
```

Comment:
This method searches through all Lessons and Homeworks to find a pair. I think, it's better to get Homework by ID and then get LessonId. Having LessonId, you can get Lesson by ID.

int LessonID = _applicationContext.Homeworks.FirstOrDefault(h => h.Id == homeworkId).LessonId;

Lesson lesson = _applicationContext.Lessons.FirstOrDefaultAsync(l =>l.Id == LessonID);

Reply:
@andy-dendy-repo you have writed the same code . i made only one Select,  you made two Select,  nothing is change .

Reply:
Try to make the next call _applicationContext.Homeworks.FirstOrDefault(h => h.Id == homeworkId).Include(Lesson).Select(homework => homework.Lesson)

Reply:
@mahotskuy  it doesnt work